DeviceBridgeTests/Controllers/TwinControllerTests.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using DeviceBridge.Models;
+using DeviceBridge.Services;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.Devices.Shared;
+using Moq;
+using NLog;
+using NUnit.Framework;
+
+namespace DeviceBridge.Controllers.Tests
+{
+    [TestFixture]
+    public class TwinControllerTests
+    {
+        private const string MockDeviceId = "test-device";
+        private const string PropertyCallbackUrl = "mock-callback-url";
+        private Mock<ISubscriptionService> _subscriptionServiceMock;
+        private Mock<IBridgeService> _bridgeServiceMock;
+        private TwinController _twinController;
+        private DeviceSubscriptionWithStatus _deviceSubscriptionWithStatus;
+
+        [SetUp]
+        public void Setup()
+        {
+            _subscriptionServiceMock = new Mock<ISubscriptionService>();
+            _bridgeServiceMock = new Mock<IBridgeService>();
+            _twinController = new TwinController(LogManager.GetCurrentClassLogger(), _subscriptionServiceMock.Object, _bridgeServiceMock.Object);
+
+            var deviceSubscription = new DeviceSubscription();
+            _deviceSubscriptionWithStatus = new DeviceSubscriptionWithStatus(deviceSubscription);
+            _subscriptionServiceMock.Setup(s => s.GetDataSubscription(It.IsAny<Logger>(), MockDeviceId, DeviceSubscriptionType.DesiredProperties, It.IsAny<CancellationToken>())).Returns(Task.FromResult(_deviceSubscriptionWithStatus));
+            _subscriptionServiceMock.Setup(s => s.CreateOrUpdateDataSubscription(It.IsAny<Logger>(), MockDeviceId, DeviceSubscriptionType.DesiredProperties, PropertyCallbackUrl, It.IsAny<CancellationToken>())).Returns(Task.FromResult(_deviceSubscriptionWithStatus));
+        }
+
+        [Test]
+        public async Task TestGetTwin()
```

Comment:
Feel our device registration should only by reported/send messages, I found "get twin" will also trigger device registration, any idea?

Reply:
Synced offline. Added a note to the README in the API section about automatic registration.